glean-core/benches/dual_labeled_counter.rs

```diff
@@ -0,0 +1,41 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use criterion::{criterion_group, criterion_main, Criterion};
+use glean_core::metrics::DualLabeledCounterMetric;
+use glean_core::CommonMetricData;
+use glean_core::Lifetime;
+use std::hint::black_box;
+
+fn bench_dual_labeled_counter_get_and_add(c: &mut Criterion) {
+    let metric = DualLabeledCounterMetric::new(
+        CommonMetricData {
+            name: "test_metric".into(),
+            category: "test_category".into(),
+            send_in_pings: vec!["test_ping".into()],
+            disabled: false,
+            lifetime: Lifetime::User,
+            ..Default::default()
+        },
+        None,
+        None,
+    );
+
+    const N: usize = 5000;
+    let keys: Vec<(String, String)> = (0..N)
+        .map(|i| (format!("key{}", i), format!("cat{}", i)))
+        .collect();
+
+    c.bench_function("dual_labeled_counter_get_and_add", |b| {
+        b.iter(|| {
+            for (k, c) in &keys {
+                let m = black_box(metric.get(k, c));
```

Comment:
This `black_box` doesn't make sense. You're using the return value, so it won't be eliminated anyway.

Reply:
Ahh, I was using it under the assumption that it would encourage the compiler to not optimize this. I'll rework this and move things to the RLB as you suggest. Thanks for the pointers here, benchmarking things like this is new territory for me :)